frontend/src/Frontend/UI/Dialogs/Send.hs

```diff
@@ -260,24 +268,57 @@ sameChainTransfer model netInfo keys fromAccount gasPayer toAccount amount = Wor
     , never
     )
 
+data InitialTransferData key
+  = InitialTransferData_Account Account
+  | InitialTransferData_Transfer (TransferData key)
+
+initialTransferDataCata :: (Account -> b) -> (TransferData key -> b) -> InitialTransferData key -> b
+initialTransferDataCata fa _ (InitialTransferData_Account a) = fa a
+initialTransferDataCata _ ft (InitialTransferData_Transfer t) = ft t
+
+-- should probably make lenses, but we'd have to move the data types out to another file because this
+-- file can't deal with the mutually recursive functions if TH was on. So lets just avoid the lenses
+
+withInitialTransfer :: (TransferData key -> a) -> InitialTransferData key -> Maybe a
+withInitialTransfer f (InitialTransferData_Transfer t)  = Just $ f t
+withInitialTransfer _ (InitialTransferData_Account _)  = Nothing
+
 -- | General transfer workflow. This is the initial configuration screen.
 sendConfig
   :: SendConstraints model mConf key t m
-  => model -> Account -> Workflow t m (mConf, Event t ())
-sendConfig model fromAccount = Workflow $ do
+  => model -> InitialTransferData key -> Workflow t m (mConf, Event t ())
+sendConfig model initData = Workflow $ do
   close <- modalHeader $ text "Send"
   rec
-    (currentTab, _done) <- makeTabs $ attachWithMaybe (const . void . hush) (current recipient) next
+    (currentTab, _done) <- makeTabs initData $ attachWithMaybe (const . void . hush) (current recipient) next
     (conf, mCaps, recipient) <- mainSection currentTab
     (cancel, next) <- footerSection currentTab recipient mCaps
   let nextScreen = flip push next $ \() -> runMaybeT $ do
-        (fromGasPayer, mToGasPayer) <- MaybeT $ sample $ current mCaps
-        (toAccount, amount) <- MaybeT $ fmap hush $ sample $ current recipient
-        let mCrossChain = ffor mToGasPayer $ \a -> CrossChainData
-              { _crossChainData_recipientChainGasPayer = a }
-        pure $ previewTransfer model fromAccount fromGasPayer toAccount mCrossChain amount
+        currTab <- lift $ sample $ current currentTab
+        if (currTab /= SendModalTab_Sign)
+          then MaybeT $ pure Nothing
+          else do
+            (fromGasPayer, mToGasPayer) <- MaybeT $ sample $ current mCaps
+            (toAccount, amount) <- MaybeT $ fmap hush $ sample $ current recipient
+            let mCrossChain = ffor mToGasPayer $ \a -> CrossChainData
+                  { _crossChainData_recipientChainGasPayer = a }
+            let transfer = TransferData
+                  { _transferData_amount = amount
+                  , _transferData_crossChainData = mCrossChain
+                  , _transferData_fromAccount = fromAccount
+                  , _transferData_fromGasPayer = fromGasPayer
+                  , _transferData_toAddress = toAccount
+                  }
```

Comment:
as `Dynamic` has a Monad instance too, you could probably avoid the explicit `sample $ current`s and have something like:
```haskell
else do
            (fromGasPayer, mToGasPayer) <- lift mCaps
            (toAccount, amount) <- lift $ fmap hush recipient
            let mCrossChain = ffor mToGasPayer $ \a -> CrossChainData
                  { _crossChainData_recipientChainGasPayer = a }
            let transfer = TransferData
                  { _transferData_amount = amount
                  , _transferData_crossChainData = mCrossChain
                  , _transferData_fromAccount = fromAccount
                  , _transferData_fromGasPayer = fromGasPayer
                  , _transferData_toAddress = toAccount
                  }
```
??

Reply:
I've heard that a puppy dies every time you use the Dynamic monad instance tho... 

The sample current stuff is not new code though, so this could be intentional. @tomsmalley ?

Reply:
No need to use the Dynamic instance, you can use the Behavior instance with `tagMaybe`. They end up doing the same thing in the end, but the tag version is a little cleaner I suppose.